File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/InvalidMessageIdException.java
package com.github.dedis.popstellar.utility.error;

import com.github.dedis.popstellar.model.network.method.message.data.Data;
import com.github.dedis.popstellar.model.objects.security.MessageID;

public class InvalidMessageIdException extends InvalidDataException {

  public InvalidMessageIdException(Data data, String id) {
    super(data, "message id", id);
  }

  public InvalidMessageIdException(Data data, MessageID id) {
    super(data, "message id", id.getEncoded());
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/InvalidMessageIdException.kt
package com.github.dedis.popstellar.utility.error

import com.github.dedis.popstellar.model.network.method.message.data.Data
import com.github.dedis.popstellar.model.objects.security.MessageID

class InvalidMessageIdException : InvalidDataException {
    constructor(data: Data, id: String) : super(data, "message id", id)
    constructor(data: Data, id: MessageID) : super(data, "message id", id.encoded)
}